Source/JavaScriptCore/API/glib/JSCDefines.h.in

```diff
@@ -47,4 +47,30 @@
 #define JSC_DEPRECATED_FOR(f) JSC_API G_DEPRECATED_FOR(f)
 #endif /* !__GI_SCANNER__ */
 
+#if ENABLE(2022_GLIB_API)
+#define JSC_DECLARE_FINAL_TYPE(ModuleObjName, module_obj_name, MODULE, OBJ_NAME, ParentName) \
+    JSC_API G_DECLARE_FINAL_TYPE(ModuleObjName, module_obj_name, MODULE, OBJ_NAME, ParentName) \
+                                                                           \
+    typedef struct _ ## ModuleObjName ## Private ModuleObjName ## Private; \
+                                                                           \
+    struct _ ## ModuleObjName {                                            \
+        ParentName parent;                                                 \
+        /*< private >*/                                                    \
+        ModuleObjName ## Private *priv;                                    \
+    };
+#else
+#define JSC_DECLARE_FINAL_TYPE(ModuleObjName, module_obj_name, MODULE, OBJ_NAME, ParentName) \
+    JSC_API GType module_obj_name ## _get_type (void);                     \
+                                                                           \
+    typedef struct _ ## ModuleObjName ModuleObjName;                       \
+    typedef struct _ ## ModuleObjName ## Class ModuleObjName ## Class;     \
+    typedef struct _ ## ModuleObjName ## Private ModuleObjName ## Private; \
+                                                                           \
+    struct _ ## ModuleObjName {                                            \
+        ParentName parent;                                                 \
+        /*< private >*/                                                    \
+        ModuleObjName ## Private *priv;                                    \
+    };
```

Comment:
Could we include the class definition here too?

Reply:
I'll try in the patch I will be splitting for the JSC changes 🫡️

Reply:
No? That would only work if every class struct is identical?

Reply:
In the case of JSC API classes, none has methods in the vtable, so we can generate them all from the macros.